th2c/stream.py

```diff
@@ -268,13 +338,16 @@ def finish(self, exc=None):
             # compose the body
             data = io.BytesIO(b''.join(self._chunks))
 
+            # extract the original request if a redirect was made
+            request = getattr(self.request, 'original_request', self.request)
```

Comment:
(edgecase-ish): If a request with multiple redirects is made, wouldn't it cause `original_request` inside the `finish` method to be the second last request made.